From dd1b1304d0b437feffd82fa676aea891e182b2a4 Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Fri, 6 Nov 2009 19:59:12 +0100
Subject: [PATCH 1/3] We are using Xkb* functions, so make sure we link against libX11

Otherwise the build fails on stricter linkers, like binutils-gold

http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=554650
---
 configure.ac                    |    1 +
 src/playground/grid/Makefile.am |    2 ++
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index 823c005..9d0cc0b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -144,6 +144,7 @@ PKG_CHECK_MODULES(GTK2, gtk+-2.0 >= $GTK2_REQUIRED)
 PKG_CHECK_MODULES(UNIQUE, unique-1.0 >= $UNIQUE_REQUIRED)
 PKG_CHECK_MODULES(LIBNOTIFY, libnotify  >= $LIBNOTIFY_REQUIRED)
 PKG_CHECK_MODULES(DEVKIT_DISKS, DeviceKit-disks  >= $DEVKIT_DISKS_REQUIRED)
+PKG_CHECK_MODULES(X11, [x11])
 
 PKG_CHECK_MODULES(LIBATASMART, [libatasmart >= 0.14])
 AC_SUBST(LIBATASMART_CFLAGS)
diff --git a/src/playground/grid/Makefile.am b/src/playground/grid/Makefile.am
index 794ea49..5ac8a35 100644
--- a/src/playground/grid/Makefile.am
+++ b/src/playground/grid/Makefile.am
@@ -31,6 +31,7 @@ grid_CFLAGS = 						\
 	$(GNOME_KEYRING_CFLAGS)				\
 	$(GTK2_CFLAGS)					\
 	$(LIBSEXY_CFLAGS)				\
+	$(X11_CFLAGS)					\
 	$(WARN_CFLAGS)					\
 	$(AM_CFLAGS)
 
@@ -45,6 +46,7 @@ grid_LDADD = 						\
 	$(GNOME_KEYRING_LIBS)				\
 	$(GTK2_LIBS)					\
 	$(LIBSEXY_LIBS)					\
+	$(X11_LIBS)					\
 	$(INTLLIBS)					\
 	$(top_builddir)/src/gdu/libgdu.la		\
 	$(top_builddir)/src/gdu-gtk/libgdu-gtk.la
-- 
1.6.5.2

